node/subsystem-util/src/lib.rs

```diff
@@ -276,33 +277,41 @@ pub fn find_validator_group(
 
 /// Choose a random subset of `min` elements.
 /// But always include `is_priority` elements.
-pub fn choose_random_subset<T, F: FnMut(&T) -> bool>(
+pub fn choose_random_subset<T, F: FnMut(&T) -> bool>(is_priority: F, v: &mut Vec<T>, min: usize) {
```

Comment:
Unless there is a good and proven perf improvement, I am not a big fan of the C-style paradigm, pass a ptr or ref and modify in that fn, I'd rather consume, modify and return.

Reply:
Well, it seems that this paradigm is more descriptive than taking `mut Vec` and it is used in many other places in Rust standard library (e.g. `sort` or even `shuffle` used in this function). I was also unsure about how to preserve `mut` in the forwarded chains (e.g. https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=6020634a0415e999e0e162f3f4f777a2). With `&mut Vec` this function does exactly what is expected: get a vector and modifies it in place.